### PR TITLE
Update to licenses/notices script

### PR DESCRIPTION
**Pull request by amzn-changml**
This updates the licenses script to pull in all PackageInfo.json files with a specific argument, then follows each license file defined and writes the contents to a file.

Also has the following features:
- Generalizes the function and variable names for non-license specific references
- Sorts `os.walk` to maintain consistent ordering
- Uses an ordered dict for the output, also to maintain ordering if using Python below 3.7
- Adds an additional json config file to have specific exclusion rules for 3p packages
- Adds a specific exclusion rule for the scanner for non-source directories (such as python). Packageinfo for these folders are still respected, however.

### CODE REVIEW

**Comment by amzn-changml** [comment]
A few issues that I'm going to correct now: 
- The ability to define multiple directories to scan
- Scan both the LicenseFile in PackageInfo.json AND look for undefined license files in the directory. Do we want to store all this in a single NOTICES.txt file or separate ones?

File: scripts/license_scanner/package_config.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "excluded_directories": [
+        ".git",
+        ".venv",
+        "build",
```
Comment (amznestebanpapp) [inline scripts/license_scanner/package_config.json:5]: recommend to also exclude the `install` folder
what about loading the exclusions from `.gitignore`?

**Review by amznestebanpapp — APPROVED** [review]

**Review by DMWeiss-divesIn — APPROVED** [review]

File: scripts/license_scanner/package_config.json (hunk 1 above)
Comment (amzn-changml) [reply scripts/license_scanner/package_config.json:5]: I was thinking about it, but need to add logic to add onto it with the excluded directories here. We don't exclude any Python folders in gitignore, for instance.

**Comment by amzn-changml** [comment]
> A few issues that I'm going to correct now:
> 
> * The ability to define multiple directories to scan, since 3p and the engine could be in completely different root paths
> * Scan both the LicenseFile in PackageInfo.json AND look for undefined license files in the directory. Do we want to store all this in a single NOTICES.txt file or separate ones?

Pushed out these changes. I ended up not using separate config files for simplicity (saw little value in that). We also decided that we will be using a single NOTICES.txt file for all scanned directories.

**Review by brianherrera — APPROVED** [review]

**Pull request merged**